ta-sdk-spi/src/main/java/com/ibm/ta/sdk/spi/plugin/TADataCollector.java

```diff
@@ -316,9 +316,14 @@ public void runAssess(PluginProvider provider, CliInputCommand cliInputCommand)
       writeRecommendationsJson(recJson, outputDir);
 
       // zip output dir
-      String zipFileName = environment.getCollectionUnitName() + ".zip";
-      File zipFile = new File(outputDir.getParentFile(), zipFileName);
-      Util.zipDir(zipFile.toPath(), outputDir);
+      // Only create zip if the data collection does not contain sensitive data
+      if (!environment.hasSensitiveData()) {
+        String zipFileName = environment.getCollectionUnitName() + ".zip";
+        File zipFile = new File(outputDir.getParentFile(), zipFileName);
+        Util.zipDir(zipFile.toPath(), outputDir);
+      } else {
+        logger.info("Data collection may contain sensitive data. No data collection zip archive will be created.");
```

Comment:
Let's be explicit here: "The enivrnoment.json file indicates that the collection contains sensitive data. No data collection zip archive will be created for collections with sensitive data